pkg/cloud/aws/actuators/cluster/actuator.go

```diff
@@ -88,6 +97,30 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 		return errors.Wrapf(err, "failed to reconcile load balancers for cluster %q", cluster.Name)
 	}
 
+	// Store KubeConfig for Cluster API NodeRef controller to use.
+	kubeConfig, err := a.Deployer.GetKubeConfig(cluster, nil)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get kubeconfig for cluster %q", cluster.Name)
+	}
+
+	kubeConfigSecret := &apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: remote.KubeConfigSecretName(cluster.Name),
+		},
+		StringData: map[string]string{
+			"value": kubeConfig,
+		},
+	}
+
+	secretClient := a.coreClient.Secrets(cluster.Namespace)
+	if _, err := secretClient.Get(kubeConfigSecret.Name, metav1.GetOptions{}); err != nil && apierrors.IsNotFound(err) {
```

Comment:
Please move this up to wrap all the secret logic

Reply:
What do you mean?

Reply:
We only need to call `a.Deployer.GetKubeConfig` and prepare an `apiv1.Secret` object if we don't find the secret when we call `Get`

Reply:
ptal